Makefile

```diff
@@ -4,18 +4,29 @@ BUILD_HASH ?= $(shell git rev-parse HEAD)
 BUILD_VERSION ?= $(shell git ls-remote --tags --refs git://github.com/mattermost/mmctl | tail -n1 | sed 's/.*\///')
 # Needed to avoid install shadow in brew which is not permitted
 ADVANCED_VET ?= TRUE
+ENTERPRISE_DIR ?= ${MM_SERVER_PATH}/../enterprise
+VENDOR_MM_SERVER_DIR ?= vendor/github.com/mattermost/mattermost-server/v5/
+ENTERPRISE_HASH ?= $(shell cat enterprise_hash)
 TESTFLAGS = -mod=vendor -timeout 30m -race -v
-
 LDFLAGS += -X "github.com/mattermost/mmctl/commands.BuildHash=$(BUILD_HASH)"
+BUILD_TAGS =
 
 .PHONY: all
 all: build
 
 -include config.override.mk
 include config.mk
 
-ifneq ($(wildcard ${MM_SERVER_PATH}/../enterprise/.*),)
-	TESTFLAGS += -ldflags '-X "github.com/mattermost/mmctl/commands.EnableEnterpriseTests=true"'
+# Prepares the enterprise build if exists. The IGNORE stuff is a hack to get the Makefile to execute the commands outside a target
+ifneq ($(wildcard ${ENTERPRISE_DIR}/.*),)
+	TESTFLAGS += -ldflags '-X "github.com/mattermost/mmctl/commands.EnableEnterpriseTests=true" -X "github.com/mattermost/mattermost-server/v5/model.BuildEnterpriseReady=true"'
+	BUILD_TAGS +=enterprise
+	IGNORE:=$(shell echo Enterprise build selected, preparing)
+	IGNORE:=$(shell rm -rf $(VENDOR_MM_SERVER_DIR)/enterprise)
+	IGNORE:=$(shell cp -R $(ENTERPRISE_DIR) $(VENDOR_MM_SERVER_DIR))
+	IGNORE:=$(shell git -C $(VENDOR_MM_SERVER_DIR)/enterprise checkout $(ENTERPRISE_HASH))
```

Comment:
I understand that the `IGNORE:=` hack is to avoid `make` stopping if the command fails, but shouldn't we fail if (for example) the enterprise directory is dirty and we fail to do the checkout?

Reply:
This was same for the server makefile, at this point it can make sense to fail because it will eventually going to fail while trying to run ee tests.